community/kernel/src/test/java/org/neo4j/test/TestSecurityManager.java

```diff
@@ -0,0 +1,440 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.test;
+
+import java.io.FileDescriptor;
+import java.net.InetAddress;
+import java.security.Permission;
+
+public class TestSecurityManager extends SecurityManager
+{
+
+    private SecurityManager securityManager;
+
+    public TestSecurityManager( SecurityManager securityManager )
+    {
+        this.securityManager = securityManager;
+    }
+
+    @Override
+    public void checkExit( int status )
+    {
+        throw new SystemExitException( status );
+    }
+
+    @Override
+    public boolean getInCheck()
+    {
+        return isManagerExist() ? securityManager.getInCheck() : super.getInCheck();
+    }
+
+    @Override
+    public Object getSecurityContext()
+    {
+        return isManagerExist() ? securityManager.getSecurityContext() : super.getSecurityContext();
+    }
+
+    @Override
+    public void checkPermission( Permission perm )
+    {
+    }
```

Comment:
Idea was to always allow everything for a test code; updated a bit to use defined securityManager if it's exist, otherwise do `noop` that will mimic `AllPermission` behaviour.
If we will call super instead of noop currently we will have a list of security exceptions since we do not have security configuration at that level atm.